src/main/scala/org/orbeon/oxf/processor/pdf/XHTMLToPDFProcessor.scala

```diff
@@ -78,57 +89,87 @@ class XHTMLToPDFProcessor() extends HttpBinarySerializer {
+      //          0.9f
+
+      //       tiz170: set callback mechanism for Flying Saucer to interact with a user agent
+      //      renderer.getSharedContext.setUserAgentCallback(
+      //        new CustomUserAgent(
+      //          jpegCompressionLevel,
+      //          pipelineContext,
+      //          renderer.getSharedContext
+      //        )
+      //      )
+
+      ////       tiz170: this is where data is passed into flying-saucer
+      //      renderer.setDocument(
+      //        readInputAsDOM(pipelineContext, input),
+      //        requestOpt map (_.getRequestURL) orNull // no base URL if can't get request URL from context
+      //      )
+      //      renderer.layout()
+
+      // tiz170: this is custom IO util
+      IOUtils.useAndClose(outputStream) { os =>
+        // tiz170 TODO: OHTP version of setting document
+        pdfRendererBuilder.toStream(os)
+        pdfRendererBuilder.withW3cDocument(
+          readInputAsDOM(pipelineContext, input),
+          requestOpt map (_.getRequestURL) orNull // no base URL if can't get request URL from context
         )
-      )
-      // renderer.getSharedContext().setDPI(150);
-
-      renderer.setDocument(
-        readInputAsDOM(pipelineContext, input),
-        requestOpt map (_.getRequestURL) orNull // no base URL if can't get request URL from context
-      )
+        //        pdfRendererBuilder.useUriResolver(new CustomFSUriResolver())
+        val pdfBoxRenderer = pdfRendererBuilder.buildPdfRenderer()
+        //         tiz170 TODO: set callback mechanism for OHTP to interact with a user agent
+        val userAgent = new CustomUserAgentOHTP(pdfBoxRenderer.getOutputDevice(), pipelineContext)
+        userAgent.setSharedContext(pdfBoxRenderer.getSharedContext)
+        pdfBoxRenderer.getSharedContext.setUserAgentCallback(userAgent)
+        pdfBoxRenderer.layout()
 
-      renderer.layout()
-
-      IOUtils.useAndClose(outputStream) { os =>
         // Page count might be zero!
         // Q: Log if no pages?
-        val hasPages = Option(renderer.getRootBox.getLayer.getPages) exists (_.size > 0)
-        if (hasPages)
-          renderer.createPDF(os)
+        // tiz170 TODO: PdfRendererBuilder has a buildPdfRenderer method which returns a PdfBoxRenderer.
+        // PdfBoxRenderer is OHTP version of ITextRenderer in flying saucer
+        //        val hasPages = Option(renderer.getRootBox.getLayer.getPages) exists (_.size > 0)
+        val hasPages = Option(pdfBoxRenderer.getRootBox.getLayer.getPages) exists (_.size > 0)
+        if (hasPages) {
+          // tiz170: this is where pdf is created using flyer saucer
+          //          renderer.createPDF(os)
+          // tiz170 TODO: add OHTP correspondence of creating PDF
+          pdfBoxRenderer.createPDF()
+        }
       }
+
     } finally {
       // Free resources associated with the rendering context
-      renderer.getSharedContext.reset()
+      //      renderer.getSharedContext.reset()
     }
```

Comment:
Here you will have to call probably the `.close()` method on `pdfBoxRenderer` to free the resources in the `finally` block.

Reply:
Thanks. Added.